File: BoundlessKit/Classes/Integration/Dashboard/BoundlessKitLauncher.m
//
//  BoundlessKitLauncher.m
//  Pods
//
//  Created by Akash Desai on 6/16/17.
//
//

#import <BoundlessKit/BoundlessKit-Swift.h>

@implementation UIApplication (BoundlessKitApplicationLauncher)

+ (void) load {
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        [[BoundlessKitApplicationLauncherBridge standard] appDidLaunch: [[NSNotification alloc] initWithName:@"" object:nil userInfo:nil]];
    });
}

@end

File: BoundlessKit/Classes/Integration/Dashboard/BoundlessKitLauncher.swift
//
//  BoundlessKitLauncher.swift
//  BoundlessKit
//
//  Created by Akash Desai on 3/7/18.
//

import Foundation

open class BoundlessKitApplicationLauncherBridge : NSObject {
    
    @objc public static let standard = BoundlessKitApplicationLauncherBridge()
    
    @objc public func appDidLaunch(_ notification: Notification) {
        // Set up boundlessKit if BoundlessProperties.plist found
        if let properties = BoundlessProperties.fromFile(using: BKUserDefaults.standard) {
            let codelessAPIClient = CodelessAPIClient(credentials: properties.credentials, version: properties.version)
            BoundlessKit._standard = BoundlessKit(apiClient: codelessAPIClient)
            codelessAPIClient.boot()
        }
    }
    
}
